### PR TITLE
fix #10 retry

### LINKED ISSUE #10: Init basic Token module

### PR DESCRIPTION
**Pull request by alex216**
fix #10 

- **First token module**
- **Fix Makefile bug**

## Summary by CodeRabbit

- **New Features**
	- Introduced a modular token management system, enhancing functionality for parsing tasks.
	- Added functions for constructing, retrieving, and destroying tokens and token lists.
	- Implemented a mechanism to manage a list of tokens effectively.

- **Bug Fixes**
	- Improved memory management to prevent leaks when creating and destroying tokens.

- **Tests**
	- Added unit tests to validate the functionality of the token management system, ensuring robustness.
	- Enhanced verification of token list operations through comprehensive testing of token insertion and retrieval.

### CODE REVIEW

**Comment by alex216** [comment]
I am mad 😠.

**Comment by alex216** [comment]
- less commit size
- `<string>` -> `<cstring>`
- func(char *, token_type) -> func(token_type, char *)
- `construct_token()` should return `t_token_list *`
    - instead of `t_token *` which is equivalent to `t_list->content`
    - be aware of wrappering
- cpp file format not working

File: Makefile [hunk 1]
```diff
@@ -20,7 +20,11 @@ LIBFT_DIR		= libft
 LIBREADLINE_DIR	= readline
 
 # Files
-SRC				= $(SRC_DIR)/main.c
+SRC				= $(SRC_DIR)/main.c \
+					$(SRC_DIR)/token/construct.c \
+					$(SRC_DIR)/token/destroy.c \
```
[the remark this answers is not in the document]
Comment (alex216) [reply Makefile:25]: `destructor.c`.

File: Makefile [hunk 2]
```diff
@@ -20,7 +20,11 @@ LIBFT_DIR		= libft
 LIBREADLINE_DIR	= readline
 
 # Files
-SRC				= $(SRC_DIR)/main.c
+SRC				= $(SRC_DIR)/main.c \
+					$(SRC_DIR)/token/construct.c \
```
[the remark this answers is not in the document]
Comment (alex216) [reply Makefile:24]: `constructor.c`.

**Review by rask24 — APPROVED** [review]
@alex216 LGTM

**Pull request merged**